integration/qat/qat_test.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+#
+# Copyright (c) 2021 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# WARNING: This script only support Intel® QAT c6xx devices
+#
+
+set -x
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o errtrace
+
+dir_path=$(dirname "$0")
+source "${dir_path}/../../lib/common.bash"
+source "${dir_path}/../../.ci/lib.sh"
+source /etc/os-release || source /usr/lib/os-release
+arch=$("${dir_path}"/../../.ci/kata-arch.sh -d)
+CI_JOB="${CI_JOB:-}"
+
+TMP_DIR=$(mktemp -d)
+QAT_DRIVER_VER=qat1.7.l.4.14.0-00031
+QAT_DRIVER_URL=https://downloadmirror.intel.com/30178/eng/${QAT_DRIVER_VER}.tar.gz
+QAT_CONF_LOCATION=${TMP_DIR}/QAT_conf
+QAT_DOCKERFILE=https://raw.githubusercontent.com/intel/intel-device-plugins-for-kubernetes/master/demo/openssl-qat-engine/Dockerfile
+QAT_SRC=${TMP_DIR}/src/QAT
+# List of QAT device IDs
+QAT_DEVICE_IDS=(37c8)
+QAT_VENDOR_ID=8086
+KATA_KERNEL_DIR=
+
+if [ "$arch" == "aarch64" ]; then
+	echo "Skip pmem test: $arch can't ensure data persistence for the lack of libpmem support"
+	exit 0
+fi
+
+if [ "${ID}" != "ubuntu" ]; then
+	echo "Skip pmem test: ${ID} distro is not supported"
+	exit 0
+fi
+
+init() {
+	local qat_devices=$(for i in ${QAT_DEVICE_IDS[*]}; do lspci -d ${QAT_VENDOR_ID}:$i; done)
+	[ -n "${qat_devices}" ] || die "This system doesn't have any QAT device"
+}
+
+build_install_qat_kernel() {
```

Comment:
@devimc, @wainersm, can we try to cache this one as we already do with the "non-qat" kernel? :-)

Reply:
@fidencio I am not sure that is possible, I believe you need special hardware for that

Reply:
it's possible but we'll need kernel sources to build the QAT modules and place it into the rootfs image, so kernel + kernel sources maybe too much to cache :man_shrugging:

Reply:
Ack.

Out of curiosity, are we adding a machine that supports QAT to our pool so we run those tests regularly?

Reply:
For now we'll testing this internally...